src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -114,6 +130,29 @@ public override TestResult Execute(TestExecutionContext context)
 
             return context.CurrentResult;
         }
+
+        private Task RunTestWithTimeoutAsync(TestExecutionContext context)
+        {
+            var testExecution = Task.Run(() => context.CurrentResult = innerCommand.Execute(context));
+
+            AwaitWithoutExceptionWrapping(testExecution, _timeout);
+
+            return testExecution;
+        }
+
+        private static void AwaitWithoutExceptionWrapping(Task<TestResult> testExecution, int timeoutMilliseconds)
+        {
+            var timeoutOrCompletion = Task
+                .WhenAny(testExecution, Task.Delay(timeoutMilliseconds))
+                .Unwrap();
```

Comment:
No action needed, but I want to comment on the preexisting code. The recommended approach for timeouts is WhenAny with `new CancellationTokenSource(timeout).Token`.

The CancellationTokenSource approach could then evolve to us using a timer just like the `THREAD_ABORT` section. When the timer ticks and debugger is not attached at that point, we call CancellationTokenSource.Cancel().